File: frameworks/csharp-dotnet/FraiseQL.Benchmark/GraphQL/Types/CommentType.cs
using FraiseQL.Benchmark.Models;

namespace FraiseQL.Benchmark.GraphQL.Types;

public class CommentType : ObjectType<Comment>
{
    protected override void Configure(IObjectTypeDescriptor<Comment> descriptor)
    {
        descriptor.Field(c => c.Id);
        descriptor.Field(c => c.Content);
        descriptor.Field(c => c.FkPost).Name("postId");
        descriptor.Field(c => c.FkAuthor).Name("authorId");
        descriptor.Field(c => c.CreatedAt);
    }
}

File: frameworks/ruby-rails/app/graphql/types/comment_type.rb
module Types
  class CommentType < Types::BaseObject
    field :id, ID, null: false
    field :content, String, null: false
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false

    field :author, Types::UserType, null: false
    field :post, Types::PostType, null: false

    def author
      User.find_by(pk_user: object.fk_author)
    end

    def post
      Post.find_by(pk_post: object.fk_post)
    end
  end
end
